src/ArborX_LinearBVH.hpp

```diff
@@ -54,8 +56,9 @@ class BoundingVolumeHierarchy
 
   bounding_volume_type bounds() const noexcept { return _bounds; }
 
-  template <typename Predicates, typename... Args>
-  void query(Predicates const &predicates, Args &&... args) const
+  template <typename ExecutionSpace, typename Predicates, typename... Args>
+  void query(ExecutionSpace const &space, Predicates const &predicates,
```

Comment:
I'm still confused why you use `A const&` here while `A&&` and `std::forward` in other places.

Reply:
I used forwarding references in the STL-like algorithms and it was a pain.  I can use it as well here, I would have to remove cvref when I check whether it is able to access the memory space of the predicates in rear case.